### PR TITLE
[KED-1963] Add option to run asynchronously via the Kedro CLI

### LINKED ISSUE #436: [KED1963] Add option to run asynchronously via the Kedro CLI
## Description
Is your feature request related to a problem? A clear and concise description of what the problem is: "I'm always frustrated when ..."

Running asynchronously should be a flag, not a context override/code change.

## Context
Why is this change important to you? How would you use it? How can it benefit other users?

Quality of life. Plus, the method shown [here](https://kedro.readthedocs.io/en/stable/04_user_guide/06_pipelines.html#asynchronous-loading-and-saving) make the `--runner`/`--parallel` options useless.

## Possible Implementation
See https://github.com/deepyaman/hookshot/commit/4cbcdb37494580c5d20bf72e211c3d9952b578c3.

### PR DESCRIPTION
**Pull request by deepyaman**
## Description

Close #436 

## Checklist

- [x] Read the [contributing](https://github.com/quantumblacklabs/kedro/blob/master/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added a description of this change and added my name to the list of supporting contributions in the [`RELEASE.md`](https://github.com/quantumblacklabs/kedro/blob/master/RELEASE.md) file
- [ ] Added tests to cover my changes

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

**Comment by 921kiyo** [comment]
@deepyaman Thank you for the PR. LGTM overall, could you please update unit tests in `tests/template/test_kedro_cli.py`, and add your name in the contributor list?

File: RELEASE.md [hunk 1]
```diff
@@ -16,6 +16,7 @@
 ## Bug fixes and other changes
 * Fixed a bug for using `ParallelRunner` on Windows.
 * Modified `GBQTableDataSet` to load customized results using customized queries from Google Big Query tables.
+* Added option to run asynchronously via the Kedro CLI.
```
Comment (lorenabalan) [inline RELEASE.md:19]: Could you move this to Upcoming Release 0.16.5? :)

**Comment by deepyaman** [comment]
Sorry for the delay, got a bit busy. I'll wrap this up shortly; just letting you know I'm on it. :)

**Comment by yetudada** [comment]
Going to close this ticket because this work is being completed in another PR.

**Pull request closed without merging**